Mathlib/Data/Set/SMulAntidiagonal.lean

```diff
@@ -0,0 +1,145 @@
+      (y.2.2.2.trans <| by
+          rw [← h]
+          exact x.2.2.2.symm).symm⟩
+
+@[to_additive VAddAntidiagonal.eq_of_fst_eq_fst]
+theorem eq_of_fst_eq_fst (h : (x : G × P).fst = (y : G × P).fst) : x = y :=
+  Subtype.ext <| Prod.ext h <| fst_eq_fst_iff_snd_eq_snd.1 h
+
+@[to_additive VAddAntidiagonal.eq_of_snd_eq_snd]
+theorem eq_of_snd_eq_snd (h : (x : G × P).snd = (y : G × P).snd) : x = y :=
+  Subtype.ext <| Prod.ext (fst_eq_fst_iff_snd_eq_snd.2 h) h
+
+end CancelSMul
+
+variable [PartialOrder G] [PartialOrder P] [SMul G P] [IsOrderedCancelSMul G P]
+  {x y : smulAntidiagonal s t a}
+
+@[to_additive VAddAntidiagonal.eq_of_fst_le_fst_of_snd_le_snd]
+theorem eq_of_fst_le_fst_of_snd_le_snd (h₁ : (x : G × P).1 ≤ (y : G × P).1)
+    (h₂ : (x : G × P).2 ≤ (y : G × P).2) : x = y :=
+  eq_of_fst_eq_fst <|
+    h₁.eq_of_not_lt fun hlt =>
+      (smul_lt_smul_of_lt_of_le hlt h₂).ne <|
+        (mem_smulAntidiagonal.1 x.2).2.2.trans (mem_smulAntidiagonal.1 y.2).2.2.symm
+
+@[to_additive VAddAntidiagonal.finite_of_isPWO]
+theorem finite_of_isPWO (hs : s.IsPWO) (ht : t.IsPWO) (a) : (smulAntidiagonal s t a).Finite := by
+  refine' Set.not_infinite.1 fun h => _
+  have h1 : (smulAntidiagonal s t a).PartiallyWellOrderedOn (Prod.fst ⁻¹'o (· ≤ ·)) := fun f hf =>
+    hs (Prod.fst ∘ f) fun n => (mem_smulAntidiagonal.1 (hf n)).1
+  have h2 : (smulAntidiagonal s t a).PartiallyWellOrderedOn (Prod.snd ⁻¹'o (· ≤ ·)) := fun f hf =>
+    ht (Prod.snd ∘ f) fun n => (mem_smulAntidiagonal.1 (hf n)).2.1
+  have isrfl : IsRefl (G × P) (Prod.fst ⁻¹'o fun x x_1 ↦ x ≤ x_1) := by
+    refine { refl := ?refl }
+    simp_all only [Order.Preimage, le_refl, Prod.forall, implies_true]
+  have istrns : IsTrans (G × P) (Prod.fst ⁻¹'o fun x x_1 ↦ x ≤ x_1) := by
+    refine { trans := ?trans }
+    simp_all only [Order.Preimage, Prod.forall]
+    exact fun a _ a_1 _ a_2 _ a_3 a_4 ↦ Preorder.le_trans a a_1 a_2 a_3 a_4
+  obtain ⟨g, hg⟩ :=
+    h1.exists_monotone_subseq (fun n => h.natEmbedding _ n) fun n => (h.natEmbedding _ n).2
+  obtain ⟨m, n, mn, h2'⟩ := h2 (fun x => (h.natEmbedding _) (g x)) fun n => (h.natEmbedding _ _).2
+  refine' mn.ne (g.injective <| (h.natEmbedding _).injective _)
+  exact eq_of_fst_le_fst_of_snd_le_snd (hg _ _ mn.le) h2'
+
+end SMulAntidiagonal
+
+@[to_additive]
+theorem Nonempty.SMul [SMul G P] {s : Set G} {t : Set P} :
+    s.Nonempty → t.Nonempty → (s • t).Nonempty :=
+  Nonempty.image2
+
+@[to_additive]
+theorem IsPWO.SMul [PartialOrder G] [PartialOrder P] [SMul G P] [IsOrderedCancelSMul G P] {s : Set G}
+    {t : Set P} (hs : s.IsPWO) (ht : t.IsPWO) : IsPWO (s • t) := by
+  rw [← @image_smul_prod]
+  exact (hs.prod ht).image_of_monotone (monotone_fst.SMul monotone_snd)
+
+@[to_additive]
+theorem IsWF.SMul [LinearOrder G] [LinearOrder P] [SMul G P] [IsOrderedCancelSMul G P] {s : Set G}
+    {t : Set P} (hs : s.IsWF) (ht : t.IsWF) : IsWF (s • t) :=
+  (hs.isPWO.SMul ht.isPWO).isWF
+
+-- _root_ seems to be needed here, and I have no idea why.
```

Comment:
It is the naming. `IsWF.SMul` now exists in the same namespace as `IsWF.min_SMul`. 

If the naming was kept as is, convention would be `IsWF.smul` but I think `Set.isWF_smul` is better. Also, for similar declarations here.